===================================================================
RCS file: /cvs/gnome/evolution-data-server/addressbook/libedata-book/e-book-backend-sexp.c,v
retrieving revision 1.15
retrieving revision 1.16
diff -u -r1.15 -r1.16
--- e-book-backend-sexp.c	2006/01/10 07:54:46	1.15
+++ addressbook/libedata-book/e-book-backend-sexp.c	2006/05/04 08:10:08	1.16
@@ -49,11 +49,14 @@
 	for (l = aims; l != NULL; l = l->next) {
 		char *im = (char *) l->data;
 
-		if (im && compare (im, str))
+		if (im && compare (im, str)) {
 			found_it = TRUE;
-
-		g_free (im);
+			break;
+		}
 	}
+	
+	g_list_foreach (aims, (GFunc)g_free, NULL);
+	g_list_free (aims);
 
 	return found_it;
 }
